hobbits/src/main/kotlin/org/apache/tuweni/hobbits/Message.kt

```diff
@@ -88,21 +71,41 @@ class Message(
    * @return the bytes of the message
    */
   fun toBytes(): Bytes {
-    val requestLine = "$protocol $version $command ${headers.size()} ${body.size()}\n"
-    return Bytes.concatenate(Bytes.wrap(requestLine.toByteArray()), headers, body)
+    val buffer = ByteBuffer.allocate(PREAMBLE.size + java.lang.Integer.BYTES * 3 + 1)
+    buffer.put(PREAMBLE)
+    buffer.putInt(version)
+    buffer.put(protocol.code)
+    buffer.putInt(headers.size())
+    buffer.putInt(body.size())
+
+    return Bytes.concatenate(Bytes.wrap(buffer.array()), headers, body)
   }
 
   /**
    * Provides the size of the message
    * @return the size of the message
    */
   fun size(): Int {
-    return protocol.length + 5 + version.length + command.length + headers.size().toString().length +
-      body.size().toString().length + headers.size() + body.size()
+    return PREAMBLE.size + java.lang.Integer.BYTES * 3 + 1 + headers.size() + body.size()
```

Comment:
just use: MESSAGE_HEADER_LENGTH instead of PREAMBLE.size + java.lang.Integer.BYTES * 3 + 1